torch/csrc/distributed/c10d/Utils.cpp

```diff
@@ -22,4 +24,29 @@ std::vector<at::Tensor> getTensorShapes(
   return shapeTensors;
 }
 
+size_t hashTensors(const std::vector<at::Tensor>& tensors) {
+  size_t hash = 0;
+  for (auto& tensor : tensors) {
+    size_t data_size = tensor.storage().nbytes();
+    std::hash<char> hasher;
+    auto src = static_cast<const char*>(tensor.storage().data_ptr().get());
+    char* dst = (char*)std::calloc(data_size, sizeof(char));
+    cudaMemcpy(dst, src, data_size, cudaMemcpyDeviceToHost);
+    for (size_t i = 0; i < data_size; ++i) {
+      // Update the hash for each byte in the tensor
+      hash ^= hasher(((char*)dst)[i]) + 0x9e3779b9 + (hash << 6) + (hash >> 2);
```

Comment:
what hasher is used here?

Reply:
We can change the hashing algorithm. I just randomly made up one here.

Reply:
well, a while back i added some hashing utils for lazy tensor that uses 128bit int https://github.com/pytorch/pytorch/blob/main/torch/csrc/lazy/core/hash.h

but there is also a hash util in c10. its probably good enough and may be faster. 
https://github.com/pytorch/pytorch/blame/main/c10/util/hash.h

might be worth comparing using one of those, but maybe yours is fine

Reply:
oh wait, I didn't see this comment. let me look into this as well.